### ISSUE TITLE
jupyter-config.py ignores path to `jupyter-proxy.pid`

### ISSUE THREAD

**Issue by MikhailSam** [jupyterhub/jupyterhub#2859]
**Describe the bug**
I'm trying to install Jupyterhub on Linux server. So I have sudo rights but I'm not a root. I already configured JupyterHub and it's config, so I can run it in single-user mode (from different folders too).

But I have this error when trying to start multi-user as described here: official documentation link.

I faced the same problem before with configurable-http-proxy and jupyterhub.sqlite - the problem was multi-user script try to save this files in system directory (`/lib/systemd/system/jupyterhub.service` or  `/etc/systemd/system/jupyterhub.service`).

I changes such parameters in jupyter_config.py:

```
## url for the database. e.g. `sqlite:///jupyterhub.sqlite`
c.JupyterHub.db_url = 'sqlite:////data/jupyterhub/jupyterhub.sqlite'

## DEPRECATED since version 0.8. Use ConfigurableHTTPProxy.command
#c.JupyterHub.proxy_cmd = []
c.ConfigurableHTTPProxy.command = '/data/anaconda3/envs/fraud/bin/configurable-http-proxy'
```
So I tried the same way for jupyterhub-proxy.pid:

```
## File to write PID Useful for daemonizing JupyterHub.
c.JupyterHub.pid_file = '/data/jupyterhub/jupyterhub-proxy.pid'
```
But looks like JupyterHub ignores it and still trying to save it to system directory! I add print to `jupyterhub/proxy.py` at `_write_pid_file(self`) function:

```
self.log.info("Writing log: %s", self.pid_file)
self.log.info("Writing log: %s", os.path.abspath(os.curdir))
```
Output:

> [I 2019-12-19 20:23:50.289 JupyterHub proxy:562] Writing proxy pid file: jupyterhub-proxy.pid
> [I 2019-12-19 20:23:50.290 JupyterHub proxy:564] Writing log: /

My idea - maybe there is another one config parameter I need to change, but I can't find anythig relevant.

**Expected behavior**
I fixed problem when by hardcode path to save the file:
```
def _write_pid_file(self):
        """write pid for proxy to a file"""
        self.log.debug("Writing proxy pid file: %s", self.pid_file)
        self.log.info("Writing proxy pid file: %s", self.pid_file)
        self.pid_file = '/data/jupyterhub/jupyterhub-proxy.pid'
        self.log.info("Writing log: %s", self.log)
        self.log.info("Writing log: %s", os.path.abspath(os.curdir))
        with open(self.pid_file, "w") as f:
            f.write(str(self.proxy_process.pid))
```

As you can see I set the same path as in the `jupyter_config.py` ! And now it works (oh, okey, I got another one error, but this is not relevant to .pid file - now JupyterHub started but I can't login :) )

**Compute Information**
 - Oracle Linux
 - JupyterHub Version 1.0.0
 - Python 3.7
 - Works in Single user

**Comment by ryogesh** [comment]
1) Can you enable debug and share the log file
2) And the jupyterhub_config file.

Scrub any sensitive information before sharing.

**Comment by willingc** [comment]
Closing this issue as it's been inactive for several months. Please feel free to ask on the [discourse site](https://discourse.jupyter.org/c/jupyterhub/10) or request that we reopen this issue. Thanks! 🌻

**Comment by atpage** [comment]
I am also having an issue with this.  `jupyterhub-proxy.pid` is written to `/` by default.  If I specify a file location with `c.JupyterHub.pid_file`, files are actually written to two locations (root, and the location I specified), containing different PIDs.

The parent PID (i.e. for `/usr/local/bin/jupyterhub`) is stored in the correct file (specified in the config), but the PID for the proxy (`/usr/local/bin/configurable-http-proxy`) is stored separately in the root directory.

So to summarize, the PID is being correctly written based on the `c.JupyterHub.pid_file` setting, but the PID of the spawned http proxy is going to `/` and doesn't seem configurable.

The solution is probably to add another setting in `jupyterhub_config.py` to specify the location for the proxy PID file.  This can then be passed to `configurable-http-proxy` with the `--pid-file` argument.
